### PR TITLE
Implements remaining changes for gustiness feature for EAMv2

### PR DESCRIPTION
**Pull request by singhbalwinder**
This PR completes the gustiness feature in EAM by implementing all the right namelist settings for gustiness. A new use case file `2000_cam5_av1c-04p2_gust.xml` has all the namelist values for invoking this feature using `FC5AV1C-L` compset. Several new namelist variables required for this feature will now appear in the namelist (`atm_in`).

A test mod directory is also created to add a test for this feature in the nightly testing.

[BFB]
[NML]

### CODE REVIEW

**Comment by singhbalwinder** [comment]
I will add a test shortly to go with this PR.

**Comment by rljacob** [comment]
@wlin7 you can start merging this.

**Comment by wlin7** [comment]
Merged to next. Conflict in tests.py was on the structure of listing the tests. Added the new test "ERS_Ld5.ne4_ne4.FC5AV1C-L.cam-gust_param" following the current format.

**Comment by rljacob** [comment]
This PR is not bfb.

**Comment by singhbalwinder** [comment]
It should be BFB. I will ask @wlin7 to merge it to next again once we isolate the NBFB change. If I can't find it today, I will revert it back.

File: components/cam/src/physics/cam/clubb_intr.F90 [hunk 1]
```diff
@@ -2500,36 +2500,37 @@ real(r8) :: ugust  ! function: gustiness as a function of convective rainfall
 
 !PMA adds gustiness and tpert
 
-    vmag_gust(:)    = 1._r8
+    vmag_gust(:)    = 0._r8
```
Comment (singhbalwinder) [inline components/cam/src/physics/cam/clubb_intr.F90]: @polunma :This is one of the changes causing NBFB behavior. Reverting it back to 1._r8 fixes it.

File: components/cam/src/control/camsrfexch.F90 [hunk 2]
```diff
@@ -465,11 +466,14 @@ subroutine cam_export(state,cam_out,pbuf)
...
    do i=1,ncol
+      umb(i)           = state%u(i,pver)
+      vmb(i)           = state%v(i,pver)
+      vmag(i)          = max(1.e-5_r8,sqrt( umb(i)**2._r8 + vmb(i)**2._r8))
       cam_out%tbot(i)  = state%t(i,pver)
       cam_out%thbot(i) = state%t(i,pver) * state%exner(i,pver)
       cam_out%zbot(i)  = state%zm(i,pver)
-      cam_out%ubot(i)  = state%u(i,pver) * vmag_gust(i)
-      cam_out%vbot(i)  = state%v(i,pver) * vmag_gust(i)
+      cam_out%ubot(i)  = state%u(i,pver) * (vmag_gust(i)+vmag(i))/vmag(i)
```
Comment (singhbalwinder) [inline components/cam/src/control/camsrfexch.F90]: @polunma : These two lines using `vmag(i)` are also causing NBFB answers.

File: components/cam/src/control/camsrfexch.F90 [hunk 3]
```diff
@@ -465,11 +466,14 @@ subroutine cam_export(state,cam_out,pbuf)
...
+      cam_out%vbot(i)  = state%v(i,pver) * (vmag_gust(i)+vmag(i))/vmag(i)
```
Comment (singhbalwinder) [inline components/cam/src/control/camsrfexch.F90]: @polunma : These two new lines using vmag(i) are also causing NBFB answers.

Comment (polunma) [reply components/cam/src/control/camsrfexch.F90]: No this won't work. I will provide a fix.

**Comment by singhbalwinder** [comment]
Thanks @polunma ! I have pushed your new fix which fixes the NBFB issue!

@wlin7 : Please merge this branch again to next.

**Comment by wlin7** [comment]
FYI: The non-BFB is due to a roundoff. If it is changed to 
     cam_out%ubot(i)  = state%u(i,pver)  * ((vmag_gust(i)+vmag(i))/vmag(i))
      cam_out%vbot(i)  = state%v(i,pver)  * ((vmag_gust(i)+vmag(i))/vmag(i))
To always compute the multiplier first before applying to the wind, it gives BFB.  This is based on SCM tests.

The if-conditional control works just as well. Any preference?

**Comment by wlin7** [comment]
Merged to next (only the updated camsrfexch.F90 involved as the previous merge was not reverted).

**Comment by wlin7** [comment]
Submit request to bless sandiatoss3 next tests:

ERS_Ld5.ne4_ne4.FC5AV1C-L.sandiatoss3_intel.cam-gust_param (new, and a baseline probably created following previous merge)

SMS_D_Ln5.ne4_ne4.FC5AV1C-L.sandiatoss3_intel.cam-cosplite_nhtfrq5 (re-bless, DIFF due to PR # 2770; it was blessed after the 1st merging of PR #2789 that contains NBFB change. This request is to reverse the DIFF due to #2789  )

For record because there were too many DIFFs yesterday due to #2789: there are two other DIFFS  but very likely not due to PR #2770.

SMS_Ln9.ne4_ne4.FC5AV1C-L.sandiatoss3_intel.cam-outfrq9s  (DIFFs for melvin next and master as well)
SMS_D_Ld1.ne30_oECv3_ICG.A_WCYCL1850S_CMIP6.sandiatoss3_intel.allactive-v1cmip6

**Comment by rljacob** [comment]
Please merge this to master.

**Comment by wlin7** [comment]
Merged to master. @singhbalwinder verified this PR does not cause the other two DIFF on sandiatoss3 (cam-outfrq9s and allactive-v1-cmip6). All NML DIFFs requested to be blessed (19 sandiatoss3, 4 melvin, 1 cori-knl).

**Pull request merged**